### ISSUE TITLE
Argument --log-level in nameko conflicts with pytest.

### ISSUE THREAD

**Issue by zhouxiaoxiang** [nameko/nameko#495]
# Argument in nameko conflicts with pytest

##  issue

- Python: 3.6.3
- pytest: 3.3.0
- nameko: 2.8.1

```

### In an empty directory, run pytest.
# cd /tmp/test
# pip install pytest
# pip install nameko
# pytest

_Traceback (most recent call last):
  File "/tmp/test/py3/bin/pytest", line 11, in <module>
    sys.exit(main())
  File "/tmp/test/py3/lib/python3.6/site-packages/_pytest/config.py", line 50, in main
    config = _prepareconfig(args, plugins)
  File "/tmp/test/py3/lib/python3.6/site-packages/_pytest/config.py", line 160, in _prepareconfig
    pluginmanager=pluginmanager, args=args)
  File "/tmp/test/py3/lib/python3.6/site-packages/pluggy/__init__.py", line 617, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/tmp/test/py3/lib/python3.6/site-packages/pluggy/__init__.py", line 222, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/tmp/test/py3/lib/python3.6/site-packages/pluggy/__init__.py", line 216, in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
  File "/tmp/test/py3/lib/python3.6/site-packages/pluggy/callers.py", line 196, in _multicall
    gen.send(outcome)
  File "/tmp/test/py3/lib/python3.6/site-packages/_pytest/helpconfig.py", line 68, in pytest_cmdline_parse
    config = outcome.get_result()
  File "/tmp/test/py3/lib/python3.6/site-packages/pluggy/callers.py", line 76, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/tmp/test/py3/lib/python3.6/site-packages/pluggy/callers.py", line 180, in _multicall
    res = hook_impl.function(*args)
  File "/tmp/test/py3/lib/python3.6/site-packages/_pytest/config.py", line 942, in pytest_cmdline_parse
    self.parse(args)
  File "/tmp/test/py3/lib/python3.6/site-packages/_pytest/config.py", line 1107, in parse
    self._preparse(args, addopts=addopts)
  File "/tmp/test/py3/lib/python3.6/site-packages/_pytest/config.py", line 1072, in _preparse
    self.known_args_namespace = ns = self._parser.parse_known_args(args, namespace=self.option.copy())
  File "/tmp/test/py3/lib/python3.6/site-packages/_pytest/config.py", line 556, in parse_known_args
    return self.parse_known_and_unknown_args(args, namespace=namespace)[0]
  File "/tmp/test/py3/lib/python3.6/site-packages/_pytest/config.py", line 562, in parse_known_and_unknown_args
    optparser = self._getparser()
  File "/tmp/test/py3/lib/python3.6/site-packages/_pytest/config.py", line 541, in _getparser
    arggroup.add_argument(*n, **a)
  File "/usr/lib/python3.6/argparse.py", line 1357, in add_argument
    return self._add_action(action)
  File "/usr/lib/python3.6/argparse.py", line 1561, in _add_action
    action = super(_ArgumentGroup, self)._add_action(action)
  File "/usr/lib/python3.6/argparse.py", line 1371, in _add_action
    self._check_conflict(action)
  File "/usr/lib/python3.6/argparse.py", line 1510, in _check_conflict
    conflict_handler(action, confl_optionals)
  File "/usr/lib/python3.6/argparse.py", line 1519, in _handle_conflict_error
    raise ArgumentError(action, message % conflict_string)
argparse.ArgumentError: argument --log-level: conflicting option string: --log-level_

```

## reason

  [**Nameko** ](https://github.com/nameko/nameko/blob/master/nameko/testing/pytest.py) and [**pytest**](https://github.com/pytest-dev/pytest/blob/master/_pytest/logging.py) have the same option --log-level for pytest_addoption, so pytest will raise argparse.ArgumentError.

**Comment by davidszotten** [comment]
ah, nice catch. for compatibility with old eventlet versions and py3, nameko pins pytest at 2.7, but only for the nameko test suite. it probably makes sense to test our "latest" with a newer pytest as well, esp since we only pin down in nameko's dev requirements

**Comment by zhouxiaoxiang** [comment]
The latest pytest  causes conflict with nameko, because:

> commit 8eafbd05ca2d980b36541fbc9d547e52b6016a9a        
> Author: Thomas Hisch <t.hisch@gmail.com>               
> Date:   Wed Sep 13 11:25:17 2017 +0200                 
> 
>     Merge the pytest-catchlog plugin                   
> 
> :040000 040000 8405b71041e3f4eee167bafb11f7407c1483a49f > > >db75ea6b6af2ec5bdcdf0887daff3a5e19285e20 M     _pytest_

```
  +    add_option_ini(
  +        parser,
  +        '--log-level',
  +        dest='log_level', default=None,
  +        help='logging level used by the logging module'
  +    )
```

  I have to pin pytest at older version temporarily if I want to use nameko and pytest together.

**Comment by askorski-fc** [comment]
I have the exact same issue:

`Traceback (most recent call last):
  File "/home/dummyuser/envs/dummyenv/bin/pytest", line 11, in <module>
    sys.exit(main())
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/_pytest/config.py", line 50, in main
    config = _prepareconfig(args, plugins)
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/_pytest/config.py", line 160, in _prepareconfig
    pluginmanager=pluginmanager, args=args)
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/pluggy/__init__.py", line 617, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/pluggy/__init__.py", line 222, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/pluggy/__init__.py", line 216, in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/pluggy/callers.py", line 196, in _multicall
    gen.send(outcome)
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/_pytest/helpconfig.py", line 68, in pytest_cmdline_parse
    config = outcome.get_result()
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/pluggy/callers.py", line 77, in get_result
    _reraise(*ex)  # noqa
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/pluggy/callers.py", line 180, in _multicall
    res = hook_impl.function(*args)
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/_pytest/config.py", line 943, in pytest_cmdline_parse
    self.parse(args)
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/_pytest/config.py", line 1108, in parse
    self._preparse(args, addopts=addopts)
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/_pytest/config.py", line 1073, in _preparse
    self.known_args_namespace = ns = self._parser.parse_known_args(args, namespace=self.option.copy())
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/_pytest/config.py", line 557, in parse_known_args
    return self.parse_known_and_unknown_args(args, namespace=namespace)[0]
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/_pytest/config.py", line 563, in parse_known_and_unknown_args
    optparser = self._getparser()
  File "/home/dummyuser/envs/dummyenv/local/lib/python2.7/site-packages/_pytest/config.py", line 542, in _getparser
    arggroup.add_argument(*n, **a)
  File "/usr/lib/python2.7/argparse.py", line 1308, in add_argument
    return self._add_action(action)
  File "/usr/lib/python2.7/argparse.py", line 1509, in _add_action
    action = super(_ArgumentGroup, self)._add_action(action)
  File "/usr/lib/python2.7/argparse.py", line 1322, in _add_action
    self._check_conflict(action)
  File "/usr/lib/python2.7/argparse.py", line 1460, in _check_conflict
    conflict_handler(action, confl_optionals)
  File "/usr/lib/python2.7/argparse.py", line 1467, in _handle_conflict_error
    raise ArgumentError(action, message % conflict_string)
argparse.ArgumentError: argument --log-level: conflicting option string(s): --log-level
`

- Python 2.7.12
- pytest 3.3.1
- nameko 2.8.1

Downgrading pytest to a version lower than 3.3 indeed works but it would be nice to fix this.

**Comment by davidszotten** [comment]
arg removed in #496 (can be restored for older pytest by installing pytest-catchlog)

**Comment by davidszotten** [comment]
released as `2.8.2`
